File: test3.go
package main;
import (
	"fmt"
	"time"
)

func primes(iterations int64) int64 {
	var primes []int64;
	var n int64;
	for n = 2; n <= iterations; n++ {
		prime := true;
		for i := 2; int64(i * i) <= n; i++ {
			if n % int64(i) == 0 {prime = false; break}
		}
		if prime {primes = append(primes, n)}
	}

	var total int64 = 0;
	for i, p := range primes {
		total += p;
		primes[i] = 0;
	}
	return total;
}

func main() {
	current := time.Now();
	result := primes(50_000_000);
	timeRequired := time.Since(current);
	fmt.Println("Sum: ", result);
	fmt.Println("Time required: ", timeRequired);
} // By Griesson

File: test3.rs
use std::time::SystemTime;

fn primes(iterations: i64) -> i64 {
    let mut primes: Vec<i32> = Vec::new();
    for n in 2..iterations {
        let mut prime = true;
        let mut i = 2;
        while i * i <= n {
            if n % i == 0 {
                prime = false;
                break;
            }
            i += 1;
        }
        if prime {primes.push(n as i32)};
    }
    let mut result: i64 = 0;
    for p in primes {
        result += p as i64;
    }
    result
}

fn main() {
    let current = SystemTime::now();
    let result = primes(50_000_000);
    let time_required = SystemTime::now().duration_since(current);
    println!("Result: {result}\nTime required: {time_required:?}");
} // By Griesson
